src/components/Sidebar/FiltersSidebar.jsx

```diff
@@ -37,27 +26,25 @@ export default function FiltersSidebar({filterOptions}) {
         }));
     };
 
-    const handleCheckboxChange = (category, index) => {
-        setFilterValues(prev => {
-            const updatedCategory = [...prev[category]];
-            updatedCategory[index].isChecked = !updatedCategory[index].isChecked;
-
-            return {
-                ...prev,
-                [category]: updatedCategory
-            };
-        });
+    const handleCheckboxChange = (category, label) => {
+        setCheckedLabels((prev) => ({
+            ...prev,
+            [category]: {
+                ...prev[category],
+                [label]: !prev[category]?.[label]
+            }
+        }));
     };
 
-    // Respond to filters changes in context
     React.useEffect( () => {
-        console.log("Filters stored ", searchTypeFilter)
-    }, [searchTypeFilter])
+        // Call endpoint to retrieve terms that match search word
+        getMatchTerms("base", "i", { filter: "", value: "" }).then(data => { 
```

Comment:
@Aiga115 why do we need this request here? we already have the terms request on the search results page, i dont think we need this request here, the filters should only update the original request that we have on the search results page, what you can do is to move this 
`<FiltersSidebar />
                     <SearchResultsBox />`
                 from the app file to a separated file, and do the fetch request that we have inside the SearchResultsBox to the new file, and to pass the results for both FiltersSidebar and SearchResultsBox